RELEASING.md

```diff
@@ -0,0 +1,19 @@
+# Releasing
+
+## Process
+
+For version v0.x.y:
+
+1. Create the annotated tag
```

Comment:
It seems like we'd be okay to use GitHub's release/tag tooling for the same effect? Is that accurate and if so should we say as much, or say why we should not?

Reply:
The idea was to try to automate the release process. Longer term, I could see a manually triggered workflow could be created to automate the tagging instead of requiring the tag to be created and pushed externally.